test/integration/setup/index.ts

```diff
@@ -147,6 +150,52 @@ export class TestEndtoEndCoordinator implements TestCoordinator {
       await proposal.simulate();
     }
 
+    if (config.category === ProposalCategory.DEBUG) {
```

Comment:
Perhaps not for this PR, but this `applyUpgrade` function is getting unwieldy and hard to follow. Think we refactor to use `switch` statements and encapsulate each `if (config.category === ProposalCategory.X)` into it's own function